commands/team_test.go

```diff
@@ -688,3 +688,219 @@ func (s *MmctlUnitTestSuite) TestSearchTeamCmd() {
+			Times(1)
+
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("private", true, "")
+
+		err := modifyTeamsCmdF(s.client, cmd, []string{"team1"})
+		s.Require().Nil(err)
+		s.Require().Equal(&mockTeam, printer.GetLines()[0])
+	})
+
+	s.Run("Modify teams, set to public", func() {
+		printer.Clean()
+		mockTeam := model.Team{
+			Id:              teamID,
+			Name:            teamName,
+			AllowOpenInvite: false,
+			Type:            model.TEAM_INVITE,
+		}
+
+		s.client.
+			EXPECT().
+			UpdateTeamPrivacy(teamID, model.TEAM_OPEN).
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("public", true, "")
+
+		err := modifyTeamsCmdF(s.client, cmd, []string{"team1"})
+		s.Require().Nil(err)
+		s.Require().Equal(&mockTeam, printer.GetLines()[0])
+	})
+
+	s.Run("Modify teams with error on UpdateTeamPrivacy returns an error", func() {
+		printer.Clean()
+		mockTeam := model.Team{
+			Id:              teamID,
+			Name:            teamName,
+			AllowOpenInvite: false,
+			Type:            model.TEAM_INVITE,
+		}
+
+		mockError := &model.AppError{
+			Message:       "An error occurred modifying a team",
+			DetailedError: "Team cannot be modified",
+			Where:         "Team.updateTeamPrivacy",
+		}
+		s.client.
+			EXPECT().
+			UpdateTeamPrivacy(teamID, model.TEAM_OPEN).
+			Return(nil, &model.Response{Error: mockError}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("public", true, "")
+
+		err := modifyTeamsCmdF(s.client, cmd, []string{"team1"})
+		s.Require().Nil(err)
+		s.Require().Equal("Unable to modify team 'team1' error: Team.updateTeamPrivacy: An error occurred modifying a team, Team cannot be modified",
+			printer.GetErrorLines()[0])
+	})
+}
+
+func (s *MmctlUnitTestSuite) TestRestoreTeamsCmd() {
+	teamName := "team1"
+	teamID := "teamId"
+	cmd := &cobra.Command{}
+
+	s.Run("Restore teams with team not exist in db returns an error", func() {
+		printer.Clean()
+
+		s.client.
+			EXPECT().
+			GetTeamByName(teamName, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		err := restoreTeamsCmdF(s.client, cmd, []string{"team1"})
+		s.Require().Nil(err)
+		s.Require().Equal("Unable to find team 'team1'", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Restore team", func() {
+		printer.Clean()
+		mockTeam := model.Team{
+			Id:   teamID,
+			Name: teamName,
+		}
+
+		s.client.
+			EXPECT().
+			RestoreTeam(teamID).
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
```

Comment:
nit: I think mocks are easier to understand if they appear in the same order they are called by the corresponding code

Reply:
Fixed here, spot below, and several other places that were pre-existing.

Reply:
Saw that you fixed it retroactively for the other PR as well, thanks a lot!! :smile: